### DISCUSSION TITLE
Get hostname of current micro?

### DISCUSSION THREAD

**Discussion by suntong** [deta/cloud-docs#559] [Help] [answered]
Is it possible to get the hostname of current micro?

Following https://stackoverflow.com/questions/7507015/get-hostname-of-current-request-in-node-js-express
I tried:

```js
var os = require("os");
os.hostname();
```

But only get _**internal**_ IP: `169.254.211.1x9`.

I need the FQDN of my micro url.

**Comment by etrotta** [comment, marked as answer]
If you are using {micro}.deta.dev, you can get the {micro} name from the [preset environment variables](https://docs.deta.sh/docs/micros/env_vars/)
Not sure if there's any way besides hardcoding for custom domains though
